addons/web/static/src/js/fields/relational_fields.js

```diff
@@ -2971,6 +2971,11 @@ var FieldSelection = AbstractField.extend({
      */
     _renderReadonly: function () {
         this.$el.empty().text(this._formatValue(this.value));
+        var raw_value = this.value;
+        if (this.field.type === 'many2one' && raw_value) {
+            raw_value = raw_value.data.id;
+        }
+        this.$el.attr('raw-value', raw_value);
```

Comment:
also, add a test for this. A single assert in an existing test or two (one for selection, one for many2one) would do the trick.